Entities/RespawningJellyfish.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Celeste.Mod.Entities;
+using Microsoft.Xna.Framework;
+using Monocle;
+using MonoMod.Utils;
+using System.Collections;
+
+namespace Celeste.Mod.SpringCollab2020.Entities {
+    [CustomEntity("SpringCollab2020/RespawningJellyfish")]
+    class RespawningJellyfish : Glider {
+        private DynData<Glider> self;
+
+        private float respawnTime;
+        private bool bubble;
+
+        private Sprite sprite;
+
+        private Vector2 initialPosition;
+        private bool respawning;
+
+        private bool shouldRespawn = true;
+
+        public RespawningJellyfish(EntityData data, Vector2 offset) : base(data, offset) {
+            respawnTime = data.Float("respawnTime");
+            bubble = data.Bool("bubble");
+            initialPosition = Position;
+            respawning = false;
+
+            // get the sprite, and give it a respawn animation.
+            self = new DynData<Glider>(this);
+            sprite = self.Get<Sprite>("sprite");
+            new DynData<Sprite>(sprite)["atlas"] = GFX.Game;

```

Comment:
I'm not 100% sure, but you should be able to `Get<Sprite>()` the sprite and update its atlas without reflection / `DynData`. Not that it really matters.